src/test/java/seedu/address/model/internship/InternshipTest.java

```diff
@@ -0,0 +1,112 @@
+                        .build();
+        assertTrue(JANESTREET.isSameInternship(editedJaneStreet));
+
+        // different companyName, all other attributes same -> returns false
+        editedJaneStreet = new InternshipBuilder(JANESTREET).withCompanyName(VALID_COMPANY_NAME_OPTIVER).build();
+        assertFalse(JANESTREET.isSameInternship(editedJaneStreet));
+
+        // companyName differs in case, all other attributes same -> returns false
+        Internship editedBob = new InternshipBuilder(OPTIVER)
+                .withCompanyName(VALID_COMPANY_NAME_OPTIVER.toLowerCase()).build();
+        assertFalse(OPTIVER.isSameInternship(editedBob));
+
+        // companyName has trailing spaces, all other attributes same -> returns false
+        String companyNameWithTrailingSpaces = VALID_COMPANY_NAME_OPTIVER + " ";
+        editedBob = new InternshipBuilder(OPTIVER).withCompanyName(companyNameWithTrailingSpaces).build();
+        assertFalse(OPTIVER.isSameInternship(editedBob));
+    }
+
+    @Test
+    public void equals() {
+        // same values -> returns true
+        Internship aliceCopy = new InternshipBuilder(JANESTREET).build();
+        assertTrue(JANESTREET.equals(aliceCopy));
+
+        // same object -> returns true
+        assertTrue(JANESTREET.equals(JANESTREET));
+
+        // null -> returns false
+        assertFalse(JANESTREET.equals(null));
+
+        // different type -> returns false
+        assertFalse(JANESTREET.equals(5));
+
+        // different internship -> returns false
+        assertFalse(JANESTREET.equals(OPTIVER));
+
+        // different companyName -> returns false
+        Internship editedJaneStreet = new InternshipBuilder(JANESTREET)
+                .withCompanyName(VALID_COMPANY_NAME_OPTIVER).build();
+        assertFalse(JANESTREET.equals(editedJaneStreet));
+
+        // different role -> returns false
+        editedJaneStreet = new InternshipBuilder(JANESTREET).withRole(VALID_ROLE_OPTIVER).build();
+        assertFalse(JANESTREET.equals(editedJaneStreet));
+
+        // different application status -> returns false
+        editedJaneStreet = new InternshipBuilder(JANESTREET)
+                .withApplicationStatus(VALID_APPLICATIONSTATUS_OPTIVER).build();
+        assertFalse(JANESTREET.equals(editedJaneStreet));
+
+        // different startDate -> returns false
+        editedJaneStreet = new InternshipBuilder(JANESTREET).withStartDate(VALID_STARTDATE_OPTIVER).build();
+        assertFalse(JANESTREET.equals(editedJaneStreet));
+
+        // different duration -> returns false
+        editedJaneStreet = new InternshipBuilder(JANESTREET).withDuration(VALID_DURATION_OPTIVER).build();
+        assertFalse(JANESTREET.equals(editedJaneStreet));
+
+        // same companyName, different role -> returns false
+        editedJaneStreet = new InternshipBuilder(JANESTREET).withRole(VALID_ROLE_OPTIVER).build();
+        assertFalse(JANESTREET.equals(editedJaneStreet));
+    }
```

Comment:
This looks like the expected output yeap. We are using the strict equality method call here instead of the isSameInternship method